runtime-modules/content/src/lib.rs

```diff
@@ -1259,8 +1307,72 @@ decl_module! {
         ) {
             Self::not_implemented()?;
         }
+
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_commitment(
+            origin,
+            new_commitment: <T as frame_system::Trait>::Hash,
+        ) {
+        ensure_root(origin)?;
+            let old_commitment = <Commitment<T>>::get();
+            if old_commitment == new_commitment { return Ok(()) }
+            <Commitment<T>>::put(new_commitment);
+            Self::deposit_event(RawEvent::CommitmentUpdated(new_commitment));
+        }
+
+    #[weight = 10_000_000]
+    pub fn update_channel_reward(
+            origin,
+            proof: PullPaymentProof<T>,
+    ) {
+
+        let signing_acc = ensure_signed(origin)?;
+        let elem = &proof.data;
+
+        let mut channel = ChannelById::<T>::get(elem.channel_id);
+
+        if let Some(channel_acc) = channel.reward_account.clone() {
+
+            let cashout = elem
+        .amount_due
+        .checked_sub(&channel.cumulative_reward)
+        .ok_or("uinteger underflow")?;
+
+            ensure!(channel_acc == signing_acc, "Origin isn't channel's reward account");
```

Comment:
You do not sign with reward account of channel, you sign with owner account of channel, check comment about extrinsic signature.